### PR TITLE
Add trigger evalation API to swagger

### PR DESCRIPTION
**Pull request by calecarter**
### Latest improvements:
Added missing triggerEvaluation API to PolicyInsights swagger
<br>
### Contribution checklist:
- [X] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [X] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [X] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json [hunk 1]
```diff
@@ -495,6 +495,75 @@
...
+          "Trigger evaluations for all resources in a subscription": {
+            "$ref": "./examples/PolicyStates_TriggerSubscriptionEvaluation.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.PolicyInsights/policyStates/latest/triggerEvaluation": {
+      "post": {
+        "operationId": "PolicyStates_TriggerResourceGroupEvaluation",
+        "description": "Triggers a policy evaluation scan for all the resources under the subscription",
```
Comment (pilor) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json]: >subscription [](start = 90, length = 12)

resource group

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json [hunk 2]
```diff
@@ -495,6 +495,75 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.PolicyInsights/policyStates/latest/triggerEvaluation": {
+      "post": {
```
Comment (pilor) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json]: >"post": { [](start = 6, length = 9)

these operations need the "x-ms-long-running-operation" tag with final-state-via = location.  [https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-long-running-operation](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-long-running-operation)

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json [hunk 3]
```diff
@@ -495,6 +495,75 @@
...
+          "202": {
+            "description": "The scan was successfully triggered.",
+            "schema": {
+              "$ref": "#/definitions/TriggerEvaluationResults"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/TriggerEvaluationFailure"
```
Comment (pilor) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json]: >TriggerEvaluationFailure [](start = 37, length = 24)

Please re-use the existing error definition for this spec "#/definitions/QueryFailure"

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json [hunk 4]
```diff
@@ -495,6 +495,75 @@
...
+            }
+          }
+        },
+        "x-ms-examples":{
+          "Trigger evaluations for all resources in a resource group": {
+            "$ref": "./examples/PolicyStates_TriggerResourceGroupEvaluation.json.json"
```
Comment (pilor) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json]: >.json.json [](start = 75, length = 10)

json inception

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json [hunk 5]
```diff
@@ -1001,6 +1070,12 @@
     }
   },
   "definitions": {
+    "TriggerEvaluationResults": {
```
Comment (pilor) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json]: >TriggerEvaluationResults [](start = 5, length = 24)

this definition needs to define its schema

Comment (calecarter) [reply specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json]: It has no schema.  I tis just an empty response with the location header defined.

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json (hunk 2 above)
Comment (calecarter) [reply specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json]: Done.

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json (hunk 1 above)
Comment (calecarter) [reply specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json]: Done

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json (hunk 3 above)
Comment (calecarter) [reply specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json]: Done

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json (hunk 4 above)
Comment (calecarter) [reply specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json]: fixed

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json (hunk 5 above)
Comment (calecarter) [reply specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json]: Just removed this definition and updated the response to not reference this.

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json [hunk 6]
```diff
@@ -495,6 +495,83 @@
...
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The scan is done."
+          },
```
Comment (sandipsh) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json]: do we get 200 or is this return code from the operation pointed by the location?

Comment (calecarter) [reply specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json]: The operation pointed by the location.  The validation tools require a terminal status code be defined.

**Review by pilor — APPROVED** [review]

**Review by sandipsh — APPROVED** [review]
:shipit:

**Comment by pilor** [comment]
@MyronFanQiu ready for you

**Review by mmyyrroonn — APPROVED** [review]

**Comment by MyronFanQiu** [comment]
@pilor @calecarter Good work. Merged it. Thanks!

**Pull request merged**